### PR TITLE
removes the tesla engine/coils from orderable cargo crates

### PR DESCRIPTION
**Pull request by SomeguyManperson**
# General Documentation

### Intent of your Pull Request

I am reducing the amount of lag from teslas by reducing the number of teslas on the station

### Why is this change good for the game?
reduces lag
# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 
tesla crate is now doesn't exist

### What general grouping does this PR fall under? 
cargo crates

# Changelog

:cl:  
rscdel: you can no longer order tesla generators or coils from cargo. This is why we can't have nice things.
rscdel: removes the cargo tesla bounty
/:cl:

### CODE REVIEW

**Comment by LazennG** [comment]
fuck teslas

**Comment by ChesterTheCheesy** [comment]
50k is a fucking overkill. Last time i was engineer, i made a triple tesla setup with around 26 tesla coils, and hooked it up to the energy harvester, which was outputting ~2,5k credits every minute. This setup took me whole 30 minutes to make. Cargo was happy, because after 4 minutes the teslas i ordered were paid off, and the rest was pure profit. However, imagine any cargo tech being told to spend 100000 for that credit generation.

10k would be somehow fine and fair. 50k just assures that nobody will ever order it, meaning you won't be able to experiment with upgrading the engine, nor doing the tesla bounty, because ~15k reward for the time and risk invested just aren't as worth anymore, considering you have to firstly have 60k to even start it.

Also, it doesn't fucking remove lag you moron, if you want lag removed from tesla, make them consume each other like singularities do, forcing people to make a single containment for each tesla.

**Comment by SomeguyManperson** [comment]
tesla price further hiked and tesla bounty removed

**Comment by ChesterTheCheesy** [comment]
> 
> 
> tesla price further hiked and tesla bounty removed

reasoning? is this PR just a shitpost?

**Comment by SomeguyManperson** [comment]
you're right I should just remove the thing entirely

**Comment by SomeguyManperson** [comment]
I was thinking I'd be able to soft cap the amount of teslas by increasing the price to the point you could only usually buy like a few but removing it solves that way better

**Comment by ChesterTheCheesy** [comment]
okay dick very good reasoning, next up i'll remove all engines and replace them with monkeys on treadmills

**Comment by TheLinuxOS** [comment]
I mean I agree that removing tesla is a good change, it has absolutely no difficulty in setting it up to the point where there are literally bounties for it. Not to mention the fact that if some idiot releases one on accident it lags the shit out of the server.

**Comment by SomeguyManperson** [comment]
if some idiot makes an engine grid out of like 20 or 40 teslas it lags the servers they don't even have to get released

**Comment by alexkar598** [comment]
No.... terrible kneejerk PR

**Comment by SomeguyManperson** [comment]
we literally reached 300% time dilation and a round nearly fucking halted because some clown wanted to make a 15 tesla engine. The tesla is in general a shit engine that requires very little thought to build or upkeep or even get power from but now it ALSO lags the shit out of the server even BEFORE getting loose
a tesla in containment invalidates the need for an engineering department aside from just fixing shit as there is no upkeep required
a tesla out of containment is an immediate shuttle call that can't be fixed in any way short of admin intervention since it nukes the entire station or leaves then nukes the entire station later
and NOW a tesla built by CARGO has been revealed to be capable of getting time dilation up so high the game nearly freezes
one tesla is enough, one tesla is MORE than enough. it's a bad engine, it's bad for the game, and it is overall just bad

**Comment by ToasterBiome** [comment]
Terrible idea. I swear half of these PRs is someone getting fucked by random thing and then bellyaching as they code out all fun in the game

**Comment by SomeguyManperson** [comment]
still haven't heard a single reason the tesla should be mass-orderable by cargo
it's too easy to set up, being a singularity but you can make it more compact since it doesn't eat stuff to the point you only need a 1x1 containment area
too easy to kill everything if it gets loose, whereas a singularity will start eating things near it giving an indicator where it is the tesla just teleports around and explodes everything
produces basically infinite power once set up with 0 management or thought required, somehow managing to be even more braindead than the antimatter which is literally just "click button whenever you expand your engine to increase power generation" by just being "click button"
impossible to re-contain if it does get loose due to the complete lack of a recontainment system and the fact that it teleports around and explodes things making it impossible to safely keep up with
and importantly a large creator of server lag, and just making the tesla explode less works for one tesla but not when you loose 15
it's like saying "hm what's the main issue with every engine we have" then smashing them all together and calling it a day, if it's in containment, it produces like five times the required amount of power with no upkeep, if it's out of containment, everything explodes, and this is only magnified by having several

**Comment by alexkar598** [comment]
Making it unspawnable outside of if you get lucky and have the right template at roundstart is against what this game stands for. We should be seeking making more things constructable, not the other way around.

Tesla also exists as the easiest engine to setup without alot of space unless you count the antimatter engine which is bullshit free power so it's very useful on those rounds where the sm delams but it doesnt quite warrent a shuttle call on it's own.


Also you know it's a fucking knee jerk pr when everything is a wall of text without proper grammar rules.

**Comment by SomeguyManperson** [comment]
not sure why I have to explain how the engine that does everything better than every other engine while also being easier to set up and having no depth in setup or maintenance is bad
the antimatter may be similarly cakewalky to set up but the tesla is also cakewalky for killing everything and still somehow requires you to press less buttons and spend less time setting it up
and every engine is "easy" to set up, at least with a basic wiki setup. The tesla however has a perfectly flat learning curve, once you set it up once that's all you need to know about it. If being "easy" is the one singular unique thing the tesla has over any other engine it's just going to be the thing people use every time since the risk/reward of its power generation is all reward and no risk
the tesla is easy to set up, it's good at generating power, it's easy to maintain, and it's good at killing everyone if it gets loose. It is bad at nothing

**Comment by JamieD1** [comment]
Can you have it locked behind CE approval or something.. or like CE has to approve. I do want cargo telsa gone but I dont want to limit people if they loose a tesla.

**Comment by Cartlord** [comment]
Bad PR, just make it so people who do dumb shit like 15-contain a Tesla get bwoinked for lagging the server
Cargo Tesla is a classic and I don't want it gone, and I don't like making it entirely RNG whether the station can tesla

**Comment by SomeguyManperson** [comment]
crate locks mean basically nothing and any approval requirement won't stop the engine from being built just make it microscopically harder.
Tesla cargo is about as "classic" as guncargo or a cargo revolt. It either does nothing or causes a massive problem some time after being started

**Comment by Cartlord** [comment]
Your entire grounds for removing this seems to be "Teslas, when contained improperly, lag the server"; have you considered that, maybe, the people who set it up in a laggy fashion should just be punished for lagging it so hard? Why punish us all for their deeds by removing tesla from crates?

**Comment by ThatOnegamer4242** [comment]
> we literally reached 300% time dilation and a round nearly fucking halted because some clown wanted to make a 15 tesla engine. The tesla is in general a shit engine that requires very little thought to build or upkeep or even get power from but now it ALSO lags the shit out of the server even BEFORE getting loose
> a tesla in containment invalidates the need for an engineering department aside from just fixing shit as there is no upkeep required
> a tesla out of containment is an immediate shuttle call that can't be fixed in any way short of admin intervention since it nukes the entire station or leaves then nukes the entire station later
> and NOW a tesla built by CARGO has been revealed to be capable of getting time dilation up so high the game nearly freezes
> one tesla is enough, one tesla is MORE than enough. it's a bad engine, it's bad for the game, and it is overall just bad

Hey uh, I'm the clown who did that. I don't think anyone here is realizing that ONE EMITTER can power the containment chamber for unlimited teslas, the issue isn't with the teslas themselves, but that the teslas are too volatile. Either they breach instantly due to incompetence or they survive forever. The fix for this is to have more emitters required to contain more teslas. That way, it takes far longer to produce a convoluted 15 tesla setup like I did. I had 15 teslas contained with 2 emitters, that's the real issue.

I would also be down with something like an anomaly neutralizer being able to kill a tesloose, making a tesloose not an instant lost cause.

**Comment by Cartlord** [comment]
> 
> 
> > we literally reached 300% time dilation and a round nearly fucking halted because some clown wanted to make a 15 tesla engine. The tesla is in general a shit engine that requires very little thought to build or upkeep or even get power from but now it ALSO lags the shit out of the server even BEFORE getting loose
> > a tesla in containment invalidates the need for an engineering department aside from just fixing shit as there is no upkeep required
> > a tesla out of containment is an immediate shuttle call that can't be fixed in any way short of admin intervention since it nukes the entire station or leaves then nukes the entire station later
> > and NOW a tesla built by CARGO has been revealed to be capable of getting time dilation up so high the game nearly freezes
> > one tesla is enough, one tesla is MORE than enough. it's a bad engine, it's bad for the game, and it is overall just bad
> 
> Hey uh, I'm the clown who did that. I don't think anyone here is realizing that ONE EMITTER can power the containment chamber for unlimited teslas, the issue isn't with the teslas themselves, but that the teslas are too volatile. Either they breach instantly due to incompetence or they survive forever. The fix for this is to have more emitters required to contain more teslas. That way, it takes far longer to produce a convoluted 15 tesla setup like I did. I had 15 teslas contained with 2 emitters, that's the real issue.
> 
> I would also be down with something like an anomaly neutralizer being able to kill a tesloose, making a tesloose not an instant lost cause.

I like everything you said here.

**Comment by Cartlord** [comment]
So this had some unintended consequences.
The "fire up a tesla" Engineering department goal is no longer attainable unless you have Tesla at roundstart.
DeltaStation is now Singulo-only because it doesn't start with a PA or Tesla engine ever. I hate Delta, yes, but I'd prefer either it be removed or people stop making it worse. 
The Tesla itself has been nerfed, and frankly I think that alone is a decent enough nerf to reduce the viability of Cargo Tesla on its own.
Also consider that someone setting tesla engine as cargo tech is INCREDIBLY BLATANT POWERGAMING, and it's not the sort of knowledge their department should have - so just AHelp them.
